airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -78,27 +89,82 @@ public CheckConnectionRead checkSourceConnection(SourceIdRequestBody sourceIdReq
     return reportConnectionStatus(schedulerJobClient.createSourceCheckConnectionJob(source, imageName));
   }
 
-  public CheckConnectionRead checkDestinationConnection(DestinationIdRequestBody destinationIdRequestBody)
+  public CheckConnectionRead checkSourceConnectionFromSourceCreate(SourceCreate sourceCreate)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    final StandardSourceDefinition sourceDef = configRepository.getStandardSourceDefinition(sourceCreate.getSourceDefinitionId());
+    final String imageName = DockerUtils.getTaggedImageName(sourceDef.getDockerRepository(), sourceDef.getDockerImageTag());
+    final SourceConnection source = new SourceConnection()
+        .withName(sourceCreate.getName())
+        .withSourceDefinitionId(sourceCreate.getSourceDefinitionId())
+        .withWorkspaceId(sourceCreate.getWorkspaceId()) // not strictly necessary.
+        .withTombstone(false)
+        // todo (cgardens) - used to create the scope so we need a value, but we are happy to have it be
+        // random.
+        .withSourceId(uuidSupplier.get())
```

Comment:
the schedulerClient relies on an id existing so that it can create a sensible scope. generating a random uuid is enough to make this work, but it's obviously a bit of hack. the alternative thing i can do here is just expose the scope / de-dupe id in this client and pipe it all the way down.